src/Tizen.TV.UIControls.Forms/Renderer/GridViewRenderer.cs

```diff
@@ -0,0 +1,246 @@
+            }
+            else if (e.PropertyName == GridView.OrientationProperty.PropertyName)
+            {
+                Control.IsHorizontal = Element.Orientation == GridViewOrientation.Horizontal ? true : false;
+            }
+            else if (e.PropertyName == GridView.ItemsSourceProperty.PropertyName)
+            {
+                UpdateItemsSource();
+            }
+            else if (e.PropertyName == GridView.SelectedItemProperty.PropertyName)
+            {
+                UpdateSelectedItem();
+            }
+            base.OnElementPropertyChanged(sender, e);
+        }
+
+        protected override void UpdateThemeStyle()
+        {
+            Control.Style = Element.ThemeStyle;
+        }
+
+        void OnItemSelected(object sender, GenGridItemEventArgs e)
+        {
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            Element.SelectedItem = context.Data;
+        }
+
+        void OnItemUnfocused(object sender, GenGridItemEventArgs e)
+        {
+            if (e.Item == null)
+                return;
+
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            (Element as IGridViewController)?.SendItemFocused(new GridViewFocusedEventArgs(context.Data, context.RealizedView, false));
+        }
+
+        void OnItemFocused(object sender, GenGridItemEventArgs e)
+        {
+            if (e.Item == null)
+                return;
+
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            (Element as IGridViewController)?.SendItemFocused(new GridViewFocusedEventArgs(context.Data, context.RealizedView, true));
+        }
+        void OnItemRealized(object sender, GenGridItemEventArgs e)
+        {
+            e.Item.EmitSignal("elm,state,5p,scale", "");
+            e.Item.SetPartColor("effect", new ElmSharp.Color(0, 0, 0, 0));
+        }
+
+        GenGridItemEventArgs OnSmartEvent(IntPtr data, IntPtr obj, IntPtr info)
+        {
+            return new GenGridItemEventArgs
+            {
+                Item = _itemContextDic.ContainsKey(info) ? _itemContextDic[info] : null
+            };
+        }
+
+        void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            UpdateItemsSource();
+        }
+
+        void UpdateItemsSource()
+        {
+            Control.Clear();
+            _gengridItemDic.Clear();
+
+            if (_collectionChanged != null)
+            {
+                _collectionChanged.CollectionChanged -= OnCollectionChanged;
+                _collectionChanged = null;
+            }
+
+            if (Element.ItemsSource == null)
+            {
+                return;
+            }
+
+            foreach (var item in Element.ItemsSource)
+            {
+                View realview = CreateView(Element.ItemTemplate, item, Element.ItemHeight, Element.ItemWidth);
+                realview.BindingContext = item;
```

Comment:
You need to consider optimize, currently, you create a view for all items, but it is not necessary,

Reply:
I fixed it. Thank you.